cloud/linode/cilium_loadbalancers.go

```diff
@@ -523,7 +524,7 @@ func (l *loadbalancers) ensureCiliumBGPPeeringPolicy(ctx context.Context) error
 		}
 	} else {
 		kv := strings.Split(Options.BGPNodeSelector, "=")
-		if len(kv) != 2 {
+		if len(kv) != maxNumNodes {
```

Comment:
This var name is incorrect. Its validating this flag input here: https://github.com/linode/linode-cloud-controller-manager/blob/main/main.go#L90